src/IECoreHoudini/DetailSplitter.cpp

```diff
@@ -165,26 +166,28 @@ DetailSplitter::Names getNames( const GU_Detail *detail )
 
 	const GA_Attribute *nameAttr = nameAttrRef.getAttribute();
 	const GA_AIFSharedStringTuple *tuple = nameAttr->getAIFSharedStringTuple();
-	GA_Size numStrings = tuple->getTableEntries( nameAttr );
 
-	results.reserve( numStrings );
-	for( GA_Size i = 0; i < numStrings; ++i )
+	for( GA_AIFSharedStringTuple::iterator it = tuple->begin( nameAttr ); !it.atEnd(); ++it )
 	{
-		GA_StringIndexType validatedIndex = tuple->validateTableHandle( nameAttr, i );
-		if( validatedIndex < 0 )
+		results.push_back( it.getString() );
+	}
+
+	GA_Range allPointsRange = detail->getPointRange();
```

Comment:
Shouldn't this be using the primRange instead since nameAttr is `GA_ATTRIB_PRIMITIVE`? Was there no performance hit for looping over all points instead of using that `validateTableHandle` method on the strings?

Reply:
Updated in 18a1733.